core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1446,22 +1456,57 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     }
   }
 
-  case object PreferredReplicaLeaderElection extends ControllerEvent {
+  def electPreferredLeaders(partitions: Set[TopicPartition], callback: (Set[TopicPartition], Map[TopicPartition, ApiError])=>Unit = { (_,_) => }): Unit =
+    eventManager.put(PreferredReplicaLeaderElection(Some(partitions), true, callback))
+
+  case class PreferredReplicaLeaderElection(partitionsOpt: Option[Set[TopicPartition]],
```

Comment:
Perhaps I misunderstand what you want, but having two different `Event` subclasses doesn't really help. It means I can potentially inline one call to `onPreferredReplicaElection()`, but otherwise just seems to result in more duplicated code. Could you explain more what you meant?